torchsde/_core/misc.py

```diff
@@ -153,3 +153,13 @@ def batch_mvp(m, v):
     mvp = torch.bmm(m, v)  # (batch_size, d, 1)
     mvp = mvp.squeeze(dim=-1)  # (batch_size, d)
     return mvp
+
+
+def grad(inputs, **kwargs):
+    # Workaround for PyTorch bug #39784
+    if torch.is_tensor(inputs):
+        inputs = (inputs,)
+    _inputs = []
```

Comment:
Nit: Could we make this a list-comprehension? Since that's usually slightly faster than a for-loop + appends. 

Also since `_input` isn't really used for subsequent computation, could we avoid defining this extra variable?

Reply:
Done the listcomp.

We have to keep `_input`. Every `input` to `autograd.grad` must have a gradient accumulator at the time that `autograd.grad` is called, which isn't true unless there's another tensor dependent on it somewhere.

Reply:
Ah that makes total sense! Just out of curiosity, from your experience with `torchdiffeq`, what were the performance differences on an average/typical setting due to this change? I guess it's clear to me now that this prevents backprop thru the graph towards the unused inputs, but would you have any estimate of the performance improvements. Again, just asking since I'm curious.

Reply:
The difference can be arbitrarily bad - and at least several orders of magnitude, i.e. seconds vs days, on problems I'm interested in.
Because autograd will run through the entire graph, and the tensor(s) being differentiated are (a function of) user-defined tensors, then in principle one can arrive at a scenario like the following:
```
x1 = torch.ones((), requires_grad=True)
x2 = torch.ones((), requires_grad=True)
x3 = torch.ones((), requires_grad=True)
y1 = x1.clone()
y2 = ArbitrarilyExpensiveFunctionToBackpropagateThrough(x2)
y = y1 + y2

torch.autograd.grad(y, (x1, x3))  # x3 is globally unused
```
In which the `y2` branch should be ignored, but because of the bug, isn't.

If there's any (admittedly unusual) custom behaviour for which repeated backwards cause different functionality then this can actually be a correctness issue as well as being a performance issue.

If you're familiar with controlled differential equations (CDEs) then the example I have in mind of is when driving one CDE with the output of another: every step in the second CDE's adjoint prompts a completely unnecessary solve of the first CDE's adjoint. (I'm happy to expand on this if you're curious, but you probably get the idea.)

Reply:
Thanks for explaining! It's good to know there are actually use cases where the difference is more than significant. Merging this PR now.